### ISSUE TITLE
Question: How to use a dynamic port and find out the port number afterwards?

### ISSUE THREAD

**Issue by jjv360** [httpswift/swifter#163]
Hi, I would be using multiple dynamically generated instances of the HttpServer. How would I find the local port that it's using?

``` swift
server = HttpServer()
server["/:path"] = shareFilesFromDirectory(bundleURL.path!)
server.start(0)
serveURL = NSURL(string: "http://localhost:" + server.port) // < How to find what port it used?
```

**Comment by nrmtrifork** [comment]
Also ports can be occupied by other applications, so it is actually a quite often task to assign port numbers to the webserver dynamically. A property like server.port and/ a callback whenever that port is set/correct would come in handy.

**Comment by glock45** [comment]
@jjv360 I need to add this. There is no method for that in Swifter.

**Comment by nrmtrifork** [comment]
That is awesome - thanks

**Comment by glock45** [comment]
The last pull requests partialy solves the problem for the dynamic allocated ports however the intention of OP was to delegate this to OS by passing "zero" as the port number (http://unix.stackexchange.com/a/180500). In that case we need to use _getsockname_ call to obtain allocated port. 

I will push a fix soon.

**Comment by mbarnach** [comment]
Fixed in commit a8e2a52.

**Comment by nrmtrifork** [comment]
This is great - thanks guys
